src/ui/components/SearchInput/index.js

```diff
@@ -13,33 +13,47 @@ export default class SearchInput extends React.Component {
     defaultValue: PropTypes.string,
     name: PropTypes.string.isRequired,
     placeholder: PropTypes.string,
+    type: PropTypes.string,
+    onChange: PropTypes.func,
   }
 
+  static defaultProps = {
+    type: 'text',
+  };
+
   constructor(props) {
     super(props);
-    this.state = { animating: false, focus: false, value: props.defaultValue };
+    this.state = { animating: false, focus: false, value: props.defaultValue || '' };
   }
 
   componentDidMount() {
     this.setIconPosition();
     window.addEventListener('resize', this.setIconPosition);
   }
 
+  componentWillReceiveProps(nextProps) {
+    const { defaultValue } = this.props;
+
+    if (defaultValue !== nextProps.defaultValue) {
+      this.setState({ value: nextProps.defaultValue || '' });
```

Comment:
One thing that might work better as `setState` is typing into a text box (like for this use case). The way React does it is every character changes the state of the component. This might be a little bit overkill for Redux since each character typed would trigger every reducer function in our app.